include/swift/AST/DiagnosticsSema.def

```diff
@@ -2847,6 +2847,15 @@ ERROR(closure_tuple_parameter_destructuring,none,
 ERROR(closure_tuple_parameter_destructuring_implicit,none,
       "closure tuple parameter %0 does not support destructuring "
       "with implicit parameters", (Type))
+ERROR(nested_tuple_parameter_destructuring,none,
+      "nested tuple parameter %0 of function %1 "
+      "does not support destructuring", (Type, Type))
+ERROR(single_tuple_parameter_mismatch,none,
+      "%0 %select{|%1 }3expects a single parameter of type %2",
```

Comment:
It's actually another way because of the condition, some of the decls don't really have names like init, deinit, subscript so we are forced do that it this way.